src/controllers/engine/colormapperjsproxy.cpp

```diff
@@ -1,20 +1,42 @@
-#include "colormapperjsproxy.h"
+#include "controllers/engine/colormapperjsproxy.h"
 
-ColorMapperJSProxy::ColorMapperJSProxy(QJSEngine* pScriptEngine, QMap<QRgb, QVariant> availableColors)
-        : m_pScriptEngine(pScriptEngine),
-          m_colorMapper(new ColorMapper(availableColors)) {
+ColorMapperJSProxy::ColorMapperJSProxy(QVariantMap availableColors)
+        : m_pColorMapper(nullptr) {
+    QMap<QRgb, QVariant> qrgbMap;
+    QVariantMap::const_iterator it;
+    for (it = availableColors.constBegin(); it != availableColors.constEnd(); ++it) {
+        QString colorString = it.key();
+        bool isInt = false;
+        if (colorString.at(0) == "#") {
+            colorString = colorString.right(colorString.size() - 1);
+            int fromHex = colorString.toInt(&isInt, 16);
+            if (isInt) {
+                qrgbMap.insert(fromHex, it.value());
+            }
+            continue;
+        }
```

Comment:
@Holzhaus do we really need this hack? We could just use numbers as the keys for the map:
```
let map = new ColorMapper({
    0xff0000: 1,
    0x00ff00: 2,
    0x0000ff: 3
});
```
If we remove the capability to use strings starting with `#` as the keys in 2.3, we don't need to keep this hack here.

Reply:
I'd be okay with that.

Reply:
removed in 5b6bea1